src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -790,6 +790,7 @@ void RadioSpinel<InterfaceType, ProcessContextType>::HandleWaitingResponse(uint3
     {
         spinel_ssize_t unpacked;
 
+        mError = OT_ERROR_NONE;
```

Comment:
Just wondering about this change and why we need it?

Reply:
When we call an RCP diag command that fails, the function `otDiagProcessCmdLine` returns an error. The RCP spinel will send the error code to the host via `SPINEL_CMD_PROP_VALUE_IS(SPINEL_PROP_LAST_STATUS)`, and the host will [store the error code](https://github.com/openthread/openthread/blob/main/src/lib/spinel/radio_spinel_impl.hpp#L786) in the variable `mError`. 
Then we call another RCP diag command successfully, the function `otDiagProcessCmdLine` returns success. The RCP spinel will send the output message to the host via `SPINEL_CMD_PROP_VALUE_IS(SPINEL_PROP_NEST_STREAM_MFG)`, and the host will parse the output message here. The previous error code stored in `mError` is not changed, then the function [PlatDiagProcess](https://github.com/openthread/openthread/blob/main/src/lib/spinel/radio_spinel_impl.hpp#L2083) returns the previous error code. It causes the RCP diag command output error message. That is not correct.

The code here is used to reset the error code.

Reply:
Make sense. SGTM. Thanks.